ops/testing.py

```diff
@@ -319,16 +319,60 @@ def set_leader(self, is_leader=True):
         if is_leader and not was_leader and self._charm is not None and self._hooks_enabled:
             self._charm.on.leader_elected.emit()
 
+    def get_backend_calls(self, reset=True):
+        """Return the calls that we have made to the TestingModelBackend.
```

Comment:
It seems a bit premature to stick that in our exposed API. We should discourage the idea of testing by checking what exact methods the implementation has called, and should instead encourage testing by checking that the expected behavior did take place.  This is strongly encouraging the former.